internal/armada/server/lease.go

```diff
@@ -676,3 +678,48 @@ func (q *AggregatedQueueServer) getJobById(jobId string) (*api.Job, error) {
 	}
 	return jobs[0], err
 }
+
+// createUsageByQueueAndExecutor Creates a map of usage first by queue and then by cluster
+// This exists because the ClusterResourceUsageReports we store in redis have to be keyed by cluster first
+// and here we  need to index by queue first.
+// Note that the desired cluster is excluded as this will be filled in later as are clusters that are not in the
+// same pool as the desired cluster
+func (q *AggregatedQueueServer) createUsageByQueueAndExecutor(reportsByCluster map[string]*schedulerobjects.ClusterResourceUsageReport, desiredCluster string, pool string) map[string]map[string]*schedulerobjects.QueueClusterResourceUsage {
+	const activeClusterExpiry = 10 * time.Minute
+	now := q.clock.Now()
+	usageByQueueAndExecutor := make(map[string]map[string]*schedulerobjects.QueueClusterResourceUsage)
+	for cluster, reports := range reportsByCluster {
+		// we don't add the reports from the desired cluster here as these will be filled in later
+		// We also don't care about any cluters that aren't in the same pool
+		if cluster != desiredCluster && reports.Pool == pool {
+			for _, report := range reports.Usages {
+				if report.Created.Add(activeClusterExpiry).After(now) {
+					if m, ok := usageByQueueAndExecutor[report.Queue]; ok {
+						m[cluster] = report
+					} else {
+						m = map[string]*schedulerobjects.QueueClusterResourceUsage{cluster: report}
+						usageByQueueAndExecutor[report.Queue] = m
+					}
+				}
+			}
+		}
+	}
+	return usageByQueueAndExecutor
+}
+
+// createClusterUsageReport creates a schedulerobjects.ClusterResourceUsageReport suitable for storing in redis
+// from the map of QueueClusterResourceUsage which is indexed by queue and cluster
+func createClusterUsageReport(reportsByQueue map[string]map[string]*schedulerobjects.QueueClusterResourceUsage, desiredCluster string, pool string) *schedulerobjects.ClusterResourceUsageReport {
```

Comment:
We can remove this now?

Reply:
We still need it (but it is much  simpler).  Leave it in for now and when e2e passes I'll integrate the new ones